File: SistemaEntregas/Auth/ProfesorFactory.cs
using SistemaEntregas.Editor;

namespace SistemaEntregas.Auth
{
    /// <summary>
    /// Concrete Factory que crea instancias de Profesor y sus componentes.
    /// </summary>
    public class ProfesorFactory : UsuarioFactory
    {
        public override Profesor CrearUsuario(string correo, string nombre, string contrasena)
        {
            var profesor = new Profesor();
            profesor.Registrar(correo, nombre, contrasena, "PROFESOR");
            return profesor;
        }

        public override EditorCodigo CrearEditor()
        {
            return new EditorCodigo
            {
                CopyPasteHabilitado = true // Profesores tienen acceso completo
            };
        }

        public override string CrearInterfaz()
        {
            return "InterfazProfesor";
        }
    }
}

File: model/Auth/ProfesorFactory.php
<?php

use model\Editor\EditorCodigo;

/**
 * Concrete Factory que crea instancias de Profesor y sus componentes.
 */
class ProfesorFactory extends UsuarioFactory
{
    public function crearUsuario(string $correo, string $nombre, string $contrasena): Profesor
    {
        $profesor = new Profesor();
        $profesor->registrar($correo, $nombre, $contrasena, 'PROFESOR');
        return $profesor;
    }

    public function crearEditor(): EditorCodigo
    {
        $editor = new EditorCodigo();
        $editor->copyPasteHabilitado = true; // Profesores tienen acceso completo
        return $editor;
    }

    public function crearInterfaz(): string
    {
        return 'InterfazProfesor';
    }
}

// Esta traducción es prácticamente idéntica a `EstudianteFactory`, 
// con las únicas diferencias propias de la lógica del original: 
// el tipo de retorno de `crearUsuario()` es `Profesor` en lugar de `Estudiante`, 
// se pasa `'PROFESOR'` al método `registrar()`, y `copyPasteHabilitado` se establece en `true` en lugar de `false`.